### PR TITLE
Branch3

### PR DESCRIPTION
**Pull request by zarembaolesia**
Added solution for homework20

### CODE REVIEW

File: src/test/java/PeopleTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import org.example.Man;
+import org.example.Woman;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class PeopleTest {
+    @Test
+    public void womanCanMarryAMan (){
```
Comment (yurii-voronenko) [inline src/test/java/PeopleTest.java:8]: Looks like method name is wrong

File: src/test/java/PeopleTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    }
+    @Test
+    public void womanChangesLastNameAfterMarryingAMan (){
+        Woman woman1 = new Woman("Jane", "Doe", 55);
+        Man man1 = new Man("John", "Wick", 50);
+
+        woman1.registerPartnership(man1);
+
+        Assert.assertEquals(woman1.getPartner().getFirstName()+woman1.getPartner().getLastName(), "John Wick", "Woman`s partner wasn`t set");
+        Assert.assertEquals(man1.getPartner().getFirstName()+man1.getPartner().getLastName(), "Jane Wick", "Man`s partner wasn`t set");
```
Comment (yurii-voronenko) [inline src/test/java/PeopleTest.java:31]: These asserts could be simplified to 
`Assert.assertEquals(woman1.getPartner(), man1);`
`Assert.assertEquals(man1.getPartner(), woman1);`

File: src/test/java/PeopleTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        Assert.assertEquals(woman1.getLastName(), "Wick", "The woman`s last name wasn`t changed after marriage");
```
Comment (yurii-voronenko) [inline src/test/java/PeopleTest.java:32]: Avoid hardcodes if it's possible
`Assert.assertEquals(woman1.getLastName(), man1.getLastName());`

File: src/test/java/PeopleTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        Assert.assertEquals(man1.getLastName(), "Wick", "The man`s last name changed after marriage");
+    }
+
+    @Test
+    public void ManCanMarryAWoman (){
```
Comment (yurii-voronenko) [inline src/test/java/PeopleTest.java:37]: method names start with low letter

File: src/test/java/PeopleTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        man3.registerPartnership(man4);
+
+        man4.deregisterPartnership(true);
+
+        Assert.assertEquals(man3.getPartner().getFirstName()+man3.getPartner().getLastName(), null, "Deregistering partnership didn`t work");
+        Assert.assertEquals(man4.getPartner().getFirstName()+man4.getPartner().getLastName(), null , "Deregistering partnership didn`t work");
+        Assert.assertEquals(man3.getLastName(), "Brown", "The man`s last name changed after divorce");
+        Assert.assertEquals(man4.getLastName(), "Johnson", "The man`s last name changed after divorce");
+    }
+}
```
Comment (yurii-voronenko) [inline src/test/java/PeopleTest.java:131]: test for retirement is absent, for man must be tested boundary values: 64, 65, 66
for woman: 59, 60, 61

**Review by yurii-voronenko — APPROVED** [review]
Done
